runner.py

```diff
@@ -17,30 +18,39 @@ def parse_manifest(text):
                   map(lambda x: x.strip(), text.splitlines()))
 
 
-def test_load(url, timeout):
-    ua_script_path = "{}/user-agent-js".format(os.getcwd())
-    test_cmd = ("timeout {timeout}s ./servo/servo '{url}'"
-                " --userscripts {ua} -x -o {png}").format(timeout=timeout,
-                                                         url=url,
-                                                         ua=ua_script_path,
-                                                         png="output.png")
-
+def execute_test(url, command, timeout):
     print("Running test:")
-    print(test_cmd)
+    print(command)
     print("Timeout:{}".format(timeout))
     try:
-        return subprocess.check_output(test_cmd, stderr=subprocess.STDOUT,
-                                      shell=True, timeout=timeout)
+        return subprocess.check_output(command, stderr=subprocess.STDOUT,
+                                       shell=True, timeout=timeout)
     except subprocess.CalledProcessError as e:
         print("Unexpected Fail:")
         print(e)
-        print("You man want to re-run the test manually:\n{}".format(test_cmd))
+        print("You may want to re-run the test manually:\n{}".format(command))
     except subprocess.TimeoutExpired:
         print("Test timeout: {}".format(url))
-
     return ""
 
 
+def run_servo(url, timeout):
+    ua_script_path = "{}/user-agent-js".format(os.getcwd())
+    test_cmd = ("timeout {timeout}s ./servo/servo '{url}'"
+                " --userscripts {ua} -x -o {png}").format(timeout=timeout,
+                                                          url=url,
+                                                          ua=ua_script_path,
+                                                          png="output.png")
+
+    return execute_test(url, test_cmd, timeout)
```

Comment:
Let's not call the `execute_test` in `run_servo`. Let's do some dependency injection like:
```
test_cmd = servo_cmd_factory(url, timeout)
execute_test(test_cmd)
```